memgpt/cli/cli.py

```diff
@@ -226,14 +211,39 @@ def quickstart(
                 with open(backup_config_path, "r", encoding="utf-8") as file:
                     backup_config = json.load(file)
                 printd("Loaded config file successfully.")
-                config_was_modified = set_config_with_dict(backup_config)
+                new_config, config_was_modified = set_config_with_dict(backup_config)
             except FileNotFoundError:
                 typer.secho(f"Config file not found at {backup_config_path}", fg=typer.colors.RED)
                 return
 
     else:
         raise NotImplementedError(backend)
 
+    if config_was_modified:
```

Comment:
Pulling the code to save the config here prevents many redundant writes to the config